software-documentation: add section, embrace manpages (#545)
void-linux/void-docs: src/config/software-documentation/index.md

> # Software documentation

Reviewer: I'm not in love with this name. Maybe `package documentation`?

Author: Done.

> Most common format of documentation in Void Linux are [manual pages](./man.md).

Reviewer: ```suggestion
The most common format of documentation in Void Linux are [manual pages](./man.md).
```

I think `is` would be more correct, but I don't know for sure.

Author: > I think `is` would be more correct

Any native speaker here?

Reviewer 2: The subject is "the most common format", so "is" would be right here, but it's still a bit awkward. I suggest one of the below alternatives:

> The most common format for documentation is the `[manual page](./man.md)`.

> The most common media for documentation are `[manual pages](./man.md)`.

Author: "Media" suggestion is neat, thanks.

> Many programs are provided along with documentation in other formats, like HTML.
> Usually it is located in `/usr/share/doc/<program>` directory.

Reviewer: ```suggestion
Many packages contain documentation in other formats, like HTML.
This documentation can usually be found in a `/usr/share/doc/<package>` directory.
```

Author: Done.

> Most extensive documentation may be split into separate `*-doc` package, like
> `julia-doc`. This is often case for programming languages, databases and big
> software libraries.

Reviewer: ```suggestion
More extensive documentation may be split into separate `*-doc` packages, such as
`julia-doc`. This is often the case for programming languages, databases and big
software libraries.
```

Author: Done.

> Software that need configuration specific to Void may include
> `/usr/share/doc/<package>/README.voidlinux` document describing it.

Reviewer: ```suggestion
Software that requires configuration specific to Void will include
`/usr/share/doc/<package>/README.voidlinux` document describing it.
```

I changed it to `will` as kind of a promise that we *will* document all Void special cases. If you don't agree with the idea or don't think it fits, feel free to remove.

That said, I don't love this paragraph. First is how it's worded, but I haven't yet figured a better way; but also that not all READMEs will be only for void specific information, since they might just include helpful information on the initial setup for some piece of software.

Author: Rewritten.
